src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -766,8 +819,9 @@ private Result<?, UpdateError> addTripToGraphAndBuffer(
       if (stopTimeUpdate.hasStopSequence()) {
         stopTime.setStopSequence(stopTimeUpdate.getStopSequence());
       }
-      stopTime.setPickupType(SCHEDULED); // Regularly scheduled pickup
-      stopTime.setDropOffType(SCHEDULED); // Regularly scheduled drop off
+      var added = AddedStopTime.ofStopTimeProperties(stopTimeUpdate.getStopTimeProperties());
+      stopTime.setPickupType(added.pickup());
+      stopTime.setDropOffType(added.dropOff());
```

Comment:
I'd check for `getScheduleRelationship` as default value (instead of previously using SCHEDULED), and only use `AddedStopTime` when extension is present. (Even if probably rare that an added trip has a stop times with 'skipped' relationship, it could happen I guess?)

Reply:
So you want to map the schedule relationship of the stop time update into the fallback for the pickup/drop off types?

Schedule relationship has the following values:
```
      SCHEDULED = 0;
      SKIPPED = 1;
      NO_DATA = 2;
      UNSCHEDULED = 3;
```

How do you suggest we map it into

```
> 0 or empty - Regularly scheduled pickup.
> 1 - No pickup available.
> 2 - Must phone agency to arrange pickup.
> 3 - Must coordinate with driver to arrange pickup.
 
?

About only using the extension when it exitsts: my goal was to abstract the extension check away and not clutter this method with it.

Do you think the check should happen here?

Reply:
Skipped and unscheduled shouldn't generate a stop times with scheduled pickup.
So everything to 0 and these two to 1 I'd say.

Reply:
Implemented in f6e76aff4db313b8b66efd1231cf60bf5775e5cb